pvlib/test/test_modelchain.py

```diff
@@ -593,29 +593,28 @@ def test_deprecated_08():
     module_parameters = {'R_sh_ref': 1, 'a_ref': 1, 'I_o_ref': 1,
                          'alpha_sc': 1, 'I_L_ref': 1, 'R_s': 1}
     # do not assign PVSystem.temperature_model_parameters
+    # without PVSystem.racking_model and PVSystem.module_type
     system = PVSystem(module_parameters=module_parameters)
-    with pytest.warns(pvlibDeprecationWarning):
-        ModelChain(system, location,
-                   dc_model='desoto',
-                   aoi_model='no_loss', spectral_model='no_loss',
-                   temp_model='sapm',
-                   ac_model='snlinverter')
-    system = PVSystem(module_parameters=module_parameters)
-    with pytest.warns(pvlibDeprecationWarning):
-        ModelChain(system, location,
-                   dc_model='desoto',
-                   aoi_model='no_loss', spectral_model='no_loss',
-                   temperature_model='sapm',
-                   temp_model='sapm',
-                   ac_model='snlinverter')
-    system = PVSystem(module_parameters=module_parameters)
-    with pytest.raises(ValueError):
-        ModelChain(system, location,
-                   dc_model='desoto',
-                   aoi_model='no_loss', spectral_model='no_loss',
-                   temperature_model='pvsyst',
-                   temp_model='sapm',
-                   ac_model='snlinverter')
+    # deprecated temp_model kwarg
+    with pytest.warns(pvlibDeprecationWarning) as recw:
+        ModelChain(system, location, dc_model='desoto', aoi_model='no_loss',
+                   spectral_model='no_loss', ac_model='snlinverter',
+                   temp_model='sapm')
+        w = recw.pop(pvlibDeprecationWarning)
+        assert 'temp_model keyword argument is deprecated' in str(w.message)
```

Comment:
all of these `asserts` need to be brought out of the context blocks.

Reply:
Working from [this](https://docs.pytest.org/en/latest/warnings.html) documentation

Reply:
I've not tried this before, but maybe it would be easier to use the `match` kwarg like

```python
    match = 'temp_model keyword argument is deprecated'
    with pytest.warns(pvlibDeprecationWarning, match=match):
        ModelChain(system, location, dc_model='desoto', aoi_model='no_loss',
                   spectral_model='no_loss', ac_model='snlinverter',
                   temp_model='sapm')
```

Reply:
`match` = regexp would probably work. The example I saw used an exact match and I didn't want to copy the exception message in full, so I stopped looking at it.  Probably `match` is better than the `in str()` pattern

Reply:
appears to work now. I think it is better to switch to using `match` rather than rely on the desired warning or exception to be the last in the stack.  Another commit coming.